scipy/stats/_binned_statistic.py

```diff
@@ -68,24 +68,41 @@ def binned_statistic(x, values, statistic='mean',
     Notes
     -----
     All but the last (righthand-most) bin is half-open.  In other words, if
-    `bins` is::
-
-      [1, 2, 3, 4]
-
-    then the first bin is ``[1, 2)`` (including 1, but excluding 2) and the
-    second ``[2, 3)``.  The last bin, however, is ``[3, 4]``, which *includes*
-    4.
+    `bins` is ``[1, 2, 3, 4]``, then the first bin is ``[1, 2)`` (including 1,
+    but excluding 2) and the second ``[2, 3)``.  The last bin, however, is
+    ``[3, 4]``, which *includes* 4.
 
     .. versionadded:: 0.11.0
 
     Examples
     --------
+    >>> from scipy import stats
+    >>> import matplotlib.pyplot as plt
+
+    First a basic example:
+
     >>> stats.binned_statistic([1, 2, 1, 2, 4], np.arange(5), statistic='mean',
-    ... bins=3)
+    ...                        bins=3)
     (array([ 1.,  2.,  4.]), array([ 1.,  2.,  3.,  4.]), array([1, 2, 1, 2, 3]))
 
-    >>> stats.binned_statistic([1, 2, 1, 2, 4], np.arange(5), statistic='mean', bins=3)
-    (array([ 1.,  2.,  4.]), array([ 1.,  2.,  3.,  4.]), array([1, 2, 1, 2, 3]))
+    Now use ``bin_edges`` and ``binnumber`` to make a plot of a distribution
+    that shows the mean and distribution around that mean per bin, on top of a
+    regular histogram and the probability distribution function:
+
+    >>> x = np.linspace(0, 5, num=500)
+    >>> x_pdf = stats.maxwell.pdf(x)
+    >>> samples = stats.maxwell.rvs(size=10000)
+
+    >>> bin_means, bin_edges, binnumber = stats.binned_statistic(x, x_pdf,
+    ...         statistic='mean', bins=25)
```

Comment:
Maybe we need to find a different application. I have looked back at where I needed this function originally. I was working out the speed that a roman warship reaches under sail for different wind speeds.
So, maybe we need some (x,y) data pairs like this:
```
# Generate some randomn data of (wind speed, boat speed) pairs
windspeed = 8 * np.random.rand(500)
boatspeed = .3 * windspeed**.5 + .2 * np.random.rand(500)

# How fast is our boat for certain certain wind speeds?
bin_means, bin_edges, binnumber = stats.binned_statistic(windspeed, boatspeed, statistic='median', bins=[1,2,3,4,5,6,7])
plt.plot(windspeed, boatspeed, 'b.', label='raw data')
plt.hlines(bin_means, bin_edges[:-1], bin_edges[1:], colors='g', lw=5, label='binned statistic of data')
plt.legend()
# Select all datapoints with a windspeed below 1
index = (binnumber==0)

```